### DISCUSSION TITLE
xdg-open from Spotify broken on NixOS

### DISCUSSION THREAD

**Discussion by noib3** [qutebrowser/qutebrowser#8319] [Q&A]
**Version info**:
<details>
  <summary>Version</summary>
qutebrowser v3.2.1
Git commit: 
Backend: QtWebEngine 6.7.2
  based on Chromium 118.0.5993.220
  with security patches up to 125.0.6422.142 (plus any distribution patches)
  (source: api)
Qt: 6.7.2

CPython: 3.12.5
PyQt: 6.7.0.dev2404081550

Qt wrapper info:
  PyQt6: success
  PyQt5: not imported
  -> selected: PyQt6 (via autoselect)

colorama: 0.4.6
jinja2: 3.1.4
pygments: 2.18.0
yaml: 6.0.2
adblock: 0.6.0
objc: no
PyQt6.QtWebEngineCore: 6.7.0
PyQt6.sip: 6.8.6
pdf.js: 4.2.67 (/nix/store/0zyi17008m7aj74w6lr1skachnqjyzrx-source/build/pdf.mjs)
sqlite: 3.46.0
QtNetwork SSL: OpenSSL 3.0.14 4 Jun 2024

Style: QFusionStyle
Platform plugin: wayland
OpenGL: AMD, 4.6 (Compatibility Profile) Mesa 24.2.2
Platform: Linux-6.11.0-x86_64-with-glibc2.39, 64bit
Linux distribution: NixOS 24.11 (Vicuna) (nixos)
Frozen: False
Imported from /nix/store/843baxq4wan14hbbmnw2yz6183394bx5-qutebrowser-3.2.1/lib/python3.12/site-packages/qutebrowser
Using Python from /nix/store/alain6ji648fj2fl073sag1324xr9ivd-python3-3.12.5-env/bin/python3.12
Qt library executable path: /nix/store/82n7plcl529pfa8ymv0gpybzx5d9zd6b-qtbase-6.7.2/libexec, data path: /nix/store/82n7plcl529pfa8ymv0gpybzx5d9zd6b-qtbase-6.7.2

Paths:
cache: /home/noib3/.cache/qutebrowser
config: /home/noib3/.config/qutebrowser
data: /home/noib3/.local/share/qutebrowser
runtime: /run/user/1000/qutebrowser
system data: /nix/store/843baxq4wan14hbbmnw2yz6183394bx5-qutebrowser-3.2.1/share/qutebrowser

Autoconfig loaded: yes
Config.py: /home/noib3/.config/qutebrowser/config.py has been loaded
Uptime: 0:07:07
</details>

**Does the bug happen if you start with `--temp-basedir`?**: N/A

**Description**

I get the following error when trying to log into Spotify's desktop app:

![2024-10-02@10:33:52](https://github.com/user-attachments/assets/3d1b38b2-c393-4bc3-a008-f434df0a9bb9)

**How to reproduce**

- set Qutebrowser as your default browser;
- install the Spotify desktop app;
- log out if you're already logged in;
- click on the green "Log in" button;

**Comment by The-Compiler** [comment]
Works fine for me on Archlinux. From what it looks like, there is nothing qutebrowser can do about this: It was launched in a misconfigured environment, where Qt cannot be loaded because apparently libproxy loads a wrong libcurl or somesuch.

Might be something NixOS-specific, where the default browser gets opened with the library overrides from Spotify or something? I found a couple of things:

- https://github.com/NixOS/nixpkgs/issues/160923 (also see the "related issues")
- https://github.com/NixOS/nixpkgs/pull/197118

So you might need to configure XDG portals on your system for xdg-open to work reliably, from what I can tell.

Converting this to a discussion in case someone else who's more familiar with NixOS has more to say about it.

**Comment by nullie** [comment]
@noib3 how do you run the Spotify app?

**Reply by noib3** [reply]
It doesn't seem to matter. I usually launch it from Fuzzel, but I get the same error if I launch it from the CLI. I also tried adding the workaround mentioned in https://github.com/NixOS/nixpkgs/pull/197118 to my home-manager config, i.e.:

```nix
    xdg.portal = {
      enable = pkgs.stdenv.isLinux;
      config.common.default = "*";
      extraPortals = with pkgs; [ xdg-desktop-portal-hyprland ];
      xdgOpenUsePortal = true;
    };
```

But that didn't help either.

**Reply by nullie** [reply]
I mean do you get it from nixpkgs, or maybe flatpak or something else?

**Reply by noib3** [reply]
Oh. Yeah from nixpkgs.
